src/blake2b.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); 
+// you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, 
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+use blake2b_simd::Params;
+
+pub fn hash(data: &[u8], output_buffer: &mut [u8]) {
+    let hash = Params::new().hash_length(output_buffer.len()).hash(data);
+    let hash_bytes = hash.as_bytes();
+    output_buffer[..hash_bytes.len()].clone_from_slice(&hash_bytes[..]);
+}
+
+#[test]
+fn test_hash_distinctness() {
+    let data_1 = b"15ed49f3b886ec5756459fdf43903c2f876ca1fb1144bb96b05e8fb116fcf7fc97fc53f1f30e350f7b304bf919b228083a8ffe1f9a0dd2162bdcb3c7a3846cc0acf04d88a5f87b32";
```

Comment:
I added a `crate::test_utils::fresh::bytestring()` that'll give you an arbitrary length bytestring that we can use here.